File: Java/7.Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int temp = x;
        long final_num = 0;
        int last = 0;
        while(temp != 0){
            last = temp % 10;
            final_num = final_num * 10 + last;
            temp = temp / 10; 
        }
        if(final_num > Integer.MAX_VALUE || final_num < Integer.MIN_VALUE){
            return 0 ;
        }
        return (int) final_num;

    }
}

File: Python/7.Reverse Integer.py
class Solution:
    def reverse(self, x: int) -> int:
        negative = False
        if(x < 0):
            negative = True
            x = abs(x)
        temp = x
        final=0
        while(temp):
            last = temp % 10 
            final = (final * 10) + last
            temp = temp // 10
        if(negative):
            final = 0 - final
        if(not (final >= (-2**31) and final <= (2**31 - 1))):
            return 0
        return final


        
        